### ISSUE TITLE
 Mitigate unstable networking on Windows pods

### ISSUE THREAD

**Issue by fujimotos** [fluent/fluent-bit#2144]
We keep hearing from our users that `filter_kubernetes` fails to start on
Windows pods. Typically, it fails with the following error.

```
[ info] [filter_kube] testing connectivity with API server...
[ warn] net_tcp_fd_connect: getaddrinfo(host='kubernetes.default.svc.cluster.local'):
        No such host is known. 
[error] [filter_kube] upstream connection error
```

This actually is k8s's fault. In particular, Windows pods are known to have
a unstable networking at boot time. This issue has been reported in
kubernetes/kubernetes/issues/78479, but left unresolved for a while.

Let's add mitigation to this issue to Fluent Bit's side, so that Fluent Bit
just works for users even when the underlying networking is unstable.

**Comment by titilambert** [comment]
Hello, I worked around this issue adding a "sleep" at the command line before starting fluentbit

**Comment by fujimotos** [comment]
> Hello, I worked around this issue adding a "sleep" at the command line before starting fluentbit

@titilambert Thank you.

Since this issue annoys so many Windows users, I decided to
implement some fix for the issue. I submitted the fix to #2186.

This patch essentially teaches Fluent Bit to be "patient" and makes
it wait until the DNS startup for you.

**Comment by fujimotos** [comment]
OK. Now #2186 is anebled Fluent Bit >= v1.4.5.
I close this ticket now.
